src/client/pfs_file.go

```
@@ -123,38 +125,35 @@ type modifyFileCore struct {
 }
 
 func (mfc *modifyFileCore) PutFile(path string, r io.Reader, opts ...PutFileOption) error {
+	config := &putFileConfig{}
+	for _, opt := range opts {
+		opt(config)
+	}
 	return mfc.maybeError(func() error {
-		pf := &pfs.PutFile{
-			Source: &pfs.PutFile_RawFileSource{
-				RawFileSource: &pfs.RawFileSource{
-					Path: path,
-				},
-			},
-		}
-		for _, opt := range opts {
-			opt(pf)
+		if !config.append {
+			if err := mfc.sendDeleteFile(&pfs.DeleteFile{
+				Path: path,
+				Tag:  config.tag,
+			}); err != nil {
+				return err
+			}
 		}
-		if err := mfc.sendPutFile(pf); err != nil {
+		if err := mfc.sendPutFile(&pfs.AddFile{
+			Path: path,
+			Tag:  config.tag,
+		}); err != nil {
 			return err
 		}
 		if _, err := grpcutil.ChunkReader(r, func(data []byte) error {
```

Nit, you can just return the err.